### PR TITLE
Trelliscope functions

### PR DESCRIPTION
**Pull request by David-Degnan**
Includes the new trelliscope functions, their test functions, and the trelliscope vignette.

### CODE REVIEW

File: vignettes/Trelliscope_Vignette_Early.Rmd [hunk 1]
```diff
@@ -0,0 +1,542 @@
...
+
+where "Input Data" is the type of data that will be plotted and "Plot Type" describes what
+kind of plot would be generated in the trelliscope. "Uses e_data" describes plots
+that can be made with just an e_data object using the as.trelliData.edata function. 
+Similarly, "Uses omicsData" describes plots that can be made with an omicsData object
+(pepData, proData, lipidData, nmrData, etc.) in the as.trelliData function. 
+"Uses statRes" describes plots that can be built with a statRes object in as.trelliData;
+however, most of these plots "Require e_meta" from omicsData to determine trelliscope
+panels. The last column, "Main Effects used" describes any plot where the "main_effects" 
+parameter in pmartR::group_designation plot was set.
```
Comment (clabornd) [inline vignettes/Trelliscope_Vignette_Early.Rmd:73]: some lingering functions/variables without back-ticks

File: R/trelliPlots.R [hunk 2]
```diff
@@ -0,0 +1,1989 @@
+# This function runs necessary checks for pmartR trelliscope plotting functions. 
+# It cleans any parameters (rounding numerics to integers, etc.), and returns them.
+trelli_precheck <- function(trelliData,
```
Comment (clabornd) [inline R/trelliPlots.R:3]: I would still document these helper functions

**Comment by clabornd** [comment]
Tests pass, package builds with remotes, vignette runs (and looks nice!).  I had some comments on documentation.

My opinion on `test_example` aside from changing the wording in the documentation:  I think it is more succinct for it to be NULL by default and do nothing, and then subset the data if user provides a vector of indices - instead of having two arguments, one's purpose of which is to activate the other. 

Not in this PR, but I was thinking about as.trelliData.edata and as.trelliData.  If all omicsData objects had a top-level "omicsData" class, this would make it possible to just have one as.trelliData function that dispatched on `data.frame` and `omicsData`?  Might be useful for other functions as well.

File: R/trelliPlots.R [hunk 3]
```diff
@@ -0,0 +1,1989 @@
...
+        panel = trelliscopejs::map2_plot(Nested_DF, as.character(unlist(toBuild[,1])), plotFUN),
+        cog = trelliscopejs::map2_cog(Nested_DF, as.character(unlist(toBuild[,1])), cogFUN)
+      ) %>%
+      trelliscopejs::trelliscope(path = path, name = name, nrow = 1, ncol = 1, thumb = T, ...) 
+    
+  }
+}
+
+# Get downloads folder
+#' @export
```
Comment (clabornd) [inline R/trelliPlots.R:211]: export needed for this function?

File: R/as.trelliData.R [hunk 4]
```diff
@@ -0,0 +1,588 @@
...
+#' @param edata Must be a dataframe. Required. 
+#' 
+#' @return A boolean where TRUE means the file is an acceptable edata file. 
+#' @examples 
+#' \dontrun{
+#' 
+#' .is_edata(pmartRdata::lipid_edata)
+#' 
+#' }
+#' @export
```
Comment (clabornd) [inline R/as.trelliData.R:14]: export wanted/necessary for this function?

**Comment by David-Degnan** [comment]
> My opinion on `test_example` aside from changing the wording in the documentation: I think it is more succinct for it to be NULL by default and do nothing, and then subset the data if user provides a vector of indices - instead of having two arguments, one's purpose of which is to activate the other.

I think we run into issues when I use the `single_plot` function with this approach, which I tried first.

**Comment by clabornd** [comment]
> I think we run into issues when I use the `single_plot` function with this approach, which I tried first.

Yea in the sense that there has to be extra logic that requires you specify single plot AND test_example or it errors?  I kind of like a solution where `single_plot = T` just chooses a random plot when `test_sample = NULL`, or the first panel when `test_sample` is a vector.  @stratkg dunno if you have an opinion either way.

File: R/trelliPlots.R (hunk 3 above)
Comment (David-Degnan) [reply R/trelliPlots.R:211]: Yes, for test functions

File: vignettes/Trelliscope_Vignette_Early.Rmd (hunk 1 above)
Comment (David-Degnan) [reply vignettes/Trelliscope_Vignette_Early.Rmd:73]: Done

File: R/trelliPlots.R (hunk 2 above)
Comment (David-Degnan) [reply R/trelliPlots.R:3]: Done

File: R/as.trelliData.R (hunk 4 above)
Comment (David-Degnan) [reply R/as.trelliData.R:14]: Yes, for test functions.

**Pull request merged**